src/api/wallet/WalletApiMock.ts

```diff
@@ -44,7 +48,31 @@ export class WalletApiMock implements WalletApi {
   public async getNetworkId(): Promise<number> {
     assert(this._connected, 'The wallet is not connected')
 
-    return Network.Rinkeby
+    return this._networkId
+  }
+
+  public addOnChangeWalletInfo(callback: (walletInfo: WalletInfo) => void, trigger?: boolean): void {
+    this._listeners.push(callback)
+    if (trigger) {
+      callback(this._getWalletInfo())
+    }
+  }
+
+  public removeOnChangeWalletInfo(callback: (walletInfo: WalletInfo) => void): void {
+    this._listeners = this._listeners.filter(c => c !== callback)
+  }
+
+  private _getWalletInfo(): WalletInfo {
+    return {
+      isConnected: this._connected,
+      userAddress: this._connected ? this._user : undefined,
+      networkId: this._connected ? this._networkId : undefined,
+    }
+  }
+
+  private async _notifyListeners(): Promise<void> {
```

Comment:
This doesn't need to be async

Reply:
Right. It was before when `this._getWalletInfo()` was async. It was using the getters instead of the attributes.

Fixed